src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -719,6 +719,7 @@ internal bool ShouldUpdateChild(TrieNode? parent, TrieNode? oldChild, TrieNode?
             if (oldChild is null && newChild is null) return false;
             if (!ReferenceEquals(oldChild, newChild)) return true;
             if (newChild.Keccak is null && parent.Keccak is not null) return true; // So that recalculate root knows to recalculate the parent root.
+            if (newChild.Keccak is null && parent.Keccak is null) return true; // parent's hash can also be null depending on nesting level - still need to update child, otherwise combine will remain original value
```

Comment:
Combine this line and the one before to just 
```suggestion
            if (newChild.Keccak is null) return true; // parent's hash can also be null depending on nesting level - still need to update child, otherwise combine will remain original value
```